benchmarks/simple.py

```diff
@@ -38,78 +51,53 @@ def loop(rc, reset_last_key=None):
     """
-    t0 = time.time()
-    for i in xrange(0, itterations):  # noqa
+    for i in range(0, num):  # noqa
         s = "foo{0}".format(i)
-
         p = rc.pipeline()
         p.set(s, i)
         p.get(s)
         p.execute()
 
-    t1 = time.time() - t0
-    print("{0}k SET/GET operations inside pipelines took: {1} seconds... {2} operations per second".format(
-        (itterations / 1000) * 2, t1, (itterations / t1) * 2)
-    )
-
 
 if __name__ == "__main__":
-    __docopt__ = """
-Usage:
-  simple [--host IP] [--port PORT] [--nocluster] [--timeit] [--pipeline] [--resetlastkey] [-h] [--version]
-
-Options:
-  --nocluster        If flag is set then StrictRedis will be used instead of cluster lib
-  --host IP          Redis server to test against [default: 127.0.0.1]
-  --port PORT        Port on redis server [default: 7000]
-  --timeit           run a mini benchmark to test performance
-  --pipeline         Only usable with --timeit flag. Runs SET/GET inside pipelines.
-  --resetlastkey     reset __last__ key
-  -h --help          show this help and exit
-  -v --version       show version and exit
-    """
-
-    args = docopt(__docopt__, version="0.3.0")
-
-    startup_nodes = [{"host": args["--host"], "port": args["--port"]}]
-
+    args = docopt(__doc__, version="0.3.1")
+    startup_nodes = [{"host": args['--host'], "port": args['--port']}]
     if not args["--nocluster"]:
         from rediscluster import StrictRedisCluster
         rc = StrictRedisCluster(startup_nodes=startup_nodes, max_connections=32, socket_timeout=0.1, decode_responses=True)
     else:
         from redis import StrictRedis
         rc = StrictRedis(host=args["--host"], port=args["--port"], socket_timeout=0.1, decode_responses=True)
-
-    if args["--timeit"]:
-        test_itterstions = [
-            5000,
-            10000,
-            20000,
-        ]
-
-        if args["--pipeline"]:
-            for itterations in test_itterstions:
-                timeit_pipeline(rc, itterations=itterations)
+    # create specified number processes
+    processes = []
+    single_request = int(args["-n"]) // int(args["-c"])
+    for j in range(int(args["-c"])):
+        if args["--timeit"]:
+            if args["--pipeline"]:
+                p = Process(target=timeit_pipeline, args=(rc, single_request))
+            else:
+                p = Process(target=timeit, args=(rc, single_request))
         else:
-            for itterations in test_itterstions:
-                timeit(rc, itterations=itterations)
-    else:
-        loop(rc, reset_last_key=args["--resetlastkey"])
+            p = Process(target=loop, args=(rc, args["--resetlastkey"]))
+        processes.append(p)
+    t1 = time.time()
+    for p in processes:
+        p.start()
+    for p in processes:
+        p.join()
+    t2 = time.time() - t1
+    print("{0}k SET/GET operations took: {1} seconds... {2} operations per second".format(int(args["-n"]) / 1000 * 2, t2, int(args["-n"]) / t2 * 2))
```

Comment:
Please remove the `*2` number inflation here. It should say something more closley to `50k SET & GET operations took...` if i specify `-n 50000`. The `-n` argument should match and output more closley to the specified value as it indicates the number of SET & GET operations that you want performed` Right now, i think there was 100k SET & GET operations done when i specify that number of itterations. Otherwise you have to change the text to something more neutral like `Tested 100k operations in ...` If you change this, remember to update the example outputs in the docs.

Reply:
how about of the commit `2e21d1419f7f614f4a9f`